File: PN18.java
import java.lang.*;
import java.util.*;

class Number
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter number : ");
        this.iNo = sobj.nextInt();
    }

    public void Display()
    {
        System.out.println("Value is : "+this.iNo);
    }

    public int CntFactor()
    {
            int iFactCnt = 1;
            int iCnt = 0;

            for(iCnt = 1; iCnt <= (iNo/2); iCnt++)
            {
                if(iNo % iCnt == 0)
				{
					iFactCnt++;
				}
            }
            return iFactCnt;
    }
}

class PN18
{
    public static void main(String b[])
    {
            Number nobj = new Number();
            int iRet = 0;

            nobj.Accept();
            nobj.Display();

            iRet =nobj.CntFactor();
            System.out.println("Factors of number are : "+iRet);
    }
}

File: PN18.cpp


#include<iostream>
using namespace std;


class Number
{
    private:
        int iNo;       

    public:
        
        void Accept()       
        {
            cout<<"Enter the value : "<<endl;
            cin>>this->iNo;
        }

        bool Display()      
        {
            cout<<"Value is : "<<this->iNo<<endl;
        }

        int SumFactors()
		{
			int iCnt = 0;
			int iSum = 0;
			
			for(iCnt = 1;iCnt <= (iNo/2); iCnt++)
			{
				if((iNo % iCnt) == 0)
				{
					iSum = iSum + iCnt;
				}
			}
			
			if(iSum == iNo)
		    {
			   return 1;
		    }
		    else
		    {
			   return 0;
		    }
		
		}
		
		

};

int main()
{
    Number nobj;
	bool iRet;

    nobj.Accept();
    nobj.Display();

    iRet = nobj.SumFactors();
	
	if(iRet == true)
	{
		cout<<"It is perfect number: ";
	}
	else
	{
		cout<<"It is not perfect number: ";
	}

    return 0;
}